src/integrationTest/java/uk/gov/hmcts/divorce/common/SubmitAosIT.java

```diff
@@ -199,18 +203,20 @@ void shouldSetStateToHoldingForValidUndisputedAosWithAosOverduePrestate() throws
         stubForIdamToken(TEST_SYSTEM_AUTHORISATION_TOKEN);
         stubForDocAssemblyWith("c35b1868-e397-457a-aa67-ac1422bb8100", "NFD_Respondent_Answers_Eng.docx");
 
-        mockMvc.perform(post(ABOUT_TO_SUBMIT_URL)
-            .contentType(APPLICATION_JSON)
-            .header(SERVICE_AUTHORIZATION, TEST_SERVICE_AUTH_TOKEN)
-            .header(AUTHORIZATION, TEST_SYSTEM_AUTHORISATION_TOKEN)
-            .content(
-                objectMapper.writeValueAsString(
-                    callbackRequest(caseData, SUBMIT_AOS, AosOverdue.name())))
-            .accept(APPLICATION_JSON))
-            .andExpect(
-                status().isOk()
-            )
-            .andExpect(jsonPath("$.state").value("Holding"));
+        for (State aosValidState : ArrayUtils.addAll(AOS_STATES, AosDrafted, AosOverdue, OfflineDocumentReceived)) {
```

Comment:
better to use Parameterized test https://github.com/hmcts/nfdiv-case-api/blob/master/src/functionalTest/java/uk/gov/hmcts/divorce/solicitor/SolicitorMidEventCallbackFT.java#L37

Reply:
updated